astropy/io/ascii/tests/test_read.py

```diff
@@ -1640,3 +1640,13 @@ def test_read_masked_bool():
     assert col.dtype.kind == 'b'
     assert np.all(col.mask == [False, False, False, True, False])
     assert np.all(col == [True, False, True, False, False])
+
+
+def test_converter_default_dict():
+    '''Check if input convertes can be default dicts.
+
+    Regression test for https://github.com/astropy/astropy/issues/4934
+    '''
+    converters = defaultdict(lambda: [ascii.convert_numpy(np.float)])
+    tab = ascii.read(['1 2'], converters=converters, format='no_header')
+    assert np.issubdtype(tab['col1'].dtype, np.floating)
```

Comment:
I do have a problem with the test, though. In it's current form, it fails for [numpy 1.20](https://github.com/astropy/astropy/pull/11889/checks?check_run_id=2900683319). If I follow the [conversion guide](https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations) and replace `np.floating` with `float`, then I get a warning which our pytest setup converts to an Exception like this:
```
E               FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.

/Users/hamogu/anaconda3/lib/python3.7/site-packages/numpy/core/numerictypes.py:422: FutureWarning
```

Any advice on how I can write a test that works for numpy <1.20 and numpy >= 1.20?

Reply:
I would have written this just using `np.float64` and then something like `assert tab['col1'].dtype.type is np.float64`. That proves that the converter resulted in exactly the type you specified.